finetuning/generalists/training/histopathology/obtain_hp_datasets.py

```diff
@@ -0,0 +1,156 @@
+
+
+"""NOTE: test sets for in-domain histopathology evaluation
+    - monuseg test split
+    - monusac test split
+    - bcss test samples (split intrinsically - in the new PR)
+
+length of individual loaders: @all (3 channel input images)
+    - lizard:  train - 718;  val - 179
+    - bcss:    train - 108;   val - 28
+    - monuseg: train - 30;   val - 7
+    - monusac: train - 168;   val - 41
+    - pannuke: train - 1294;  val - 680
+"""
+
+
+def _get_train_val_split(ds, val_fraction: float = 0.2):
+    generator = torch.Generator().manual_seed(42)
+    train_ds, val_ds = data_util.random_split(ds, [1 - val_fraction, val_fraction], generator=generator)
+    return train_ds, val_ds
+
+
+class BCSSLabelTrafo:
+    def __init__(self, label_choices: Optional[List[int]] = None, do_connected_components: bool = False):
+        self.label_choices = label_choices
+        self.do_connected_components = do_connected_components
+
+    def __call__(self, labels: np.ndarray) -> np.ndarray:
+        """Returns the transformed bcss data labels (use-case for SAM)"""
+        if self.label_choices is not None:
+            labels[~np.isin(labels, self.label_choices)] = 0
+
+        if self.do_connected_components:
+            segmentation = measure.label(labels)
+        else:
+            segmentation = label_padding_trafo(labels)
+
+        return segmentation
+
+
+def raw_padding_trafo(raw, desired_shape=(3, 512, 512)):
+    assert raw.shape[0] == 3, "The input shape isn't channels first, expected: (3, H, W)"
+    tmp_ddim = (desired_shape[1] - raw.shape[1], desired_shape[2] - raw.shape[2])
+    ddim = (tmp_ddim[0] / 2, tmp_ddim[1] / 2)
+    raw = np.pad(
+        raw,
+        pad_width=((0, 0), (ceil(ddim[0]), floor(ddim[0])), (ceil(ddim[1]), floor(ddim[1]))),
+        mode="reflect"
+    )
+    assert raw.shape == desired_shape
+    return raw
+
+
+def label_padding_trafo(labels, desired_shape=(512, 512)):
+    tmp_ddim = (desired_shape[0] - labels.shape[0], desired_shape[1] - labels.shape[1])
+    ddim = (tmp_ddim[0] / 2, tmp_ddim[1] / 2)
+    labels = np.pad(
+        labels,
+        pad_width=((ceil(ddim[0]), floor(ddim[0])), (ceil(ddim[1]), floor(ddim[1]))),
+        mode="reflect"
+    )
+    assert labels.shape == desired_shape
+    labels = label_consecutive_trafo(labels)
+    return labels
+
+
+def label_consecutive_trafo(labels):
+    labels = labels.astype(int)
+    labels = torch_em.transform.label.label_consecutive(labels)  # to ensure consecutive IDs
+    return labels
+
+
+def get_concat_hp_datasets(path, patch_shape):
+    label_dtype = torch.int64
+    sampler = MinInstanceSampler(min_num_instances=5)
+
+    # make lizard dataset splits into fractions
+    lizard_ds = datasets.get_lizard_dataset(
+        path=os.path.join(path, "lizard"), patch_shape=patch_shape, sampler=sampler, label_dtype=label_dtype,
+        raw_transform=raw_padding_trafo, label_transform=label_padding_trafo
```

Comment:
Using the `raw_padding_trafo` like this will remove data normalization from the data loader. I am not sure if this leads to any problems.